### PR TITLE
Physics

### LINKED ISSUE #34: Collision detection
There needs to be collision detection to ensure that the tanks cannot run through each other and that when a bullet hits a tank that it registers with the game that that occurred.

### LINKED ISSUE #38: Damage detection
When a bullet hits a tank then the tank should suffer damage.

### PR DESCRIPTION
**Pull request by Cousjava**
Adds basic collision detection so tanks don't go through each other, and adds walls to prevent tanks going offscreen.

Closes #34 
Closes #38
Closes #52

### CODE REVIEW

**Comment by Cousjava** [comment]
closes #34

File: core/src/main/java/com/github/integritygame/screens/MainGameScreen.java [hunk 1]
```diff
@@ -21,6 +21,9 @@
      * Create a main game screen and initialise gamemanager to handle most the stuff
      */
     public MainGameScreen(){
+        spriteBatch = new SpriteBatch();
```
Comment (Zach-S1) [inline core/src/main/java/com/github/integritygame/screens/MainGameScreen.java]: This is already initialised

File: core/src/main/java/com/github/integritygame/screens/MainGameScreen.java [hunk 2]
```diff
@@ -21,6 +21,9 @@
...
+        shapeRenderer = new ShapeRenderer();
```
Comment (Zach-S1) [inline core/src/main/java/com/github/integritygame/screens/MainGameScreen.java]: This is already initialised as well

File: desktop/build.gradle [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apply plugin: "java"
 
 sourceCompatibility = 1.8
-sourceSets.main.java.srcDirs = [ "src/main/java/" ]
+sourceSets.main.java.srcDirs = [ "src/main" ]
```
Comment (Zach-S1) [inline desktop/build.gradle]: This needs to stay as src/main/java/ as that is the source dir

File: desktop/src/main/java/com/github/integritygame/desktop/DesktopLauncher.java [hunk 4]
```diff
@@ -7,10 +7,6 @@
 public class DesktopLauncher {
 	public static void main (String[] arg) {
 		LwjglApplicationConfiguration config = new LwjglApplicationConfiguration();
-			config.title = "Tank Game";
```
Comment (Zach-S1) [inline desktop/src/main/java/com/github/integritygame/desktop/DesktopLauncher.java]: I don't think these need deleting as we want the game to stay the same size as we don't want it resizable yet.

File: core/src/main/java/com/github/integritygame/objects/Bullet.java [hunk 5]
```diff
@@ -49,4 +51,55 @@ public int getTextureHeight(){
...
+    void setBody(Body bulletBody) {
+        body = bulletBody;
+        body.setUserData(this);
+
+    }
+    
+    /**
+     * Returns the amount of damage that is dealt to a
+     * tank on impact
+     */
```
Comment (jwilliamson15) [inline core/src/main/java/com/github/integritygame/objects/Bullet.java]: javadoc need @return line

File: core/src/main/java/com/github/integritygame/objects/Bullet.java [hunk 6]
```diff
@@ -49,4 +51,55 @@ public int getTextureHeight(){
...
+        if (!impacted) {
+            return damage;
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Whether the bullet has hit another object
+     * @return
```
Comment (jwilliamson15) [inline core/src/main/java/com/github/integritygame/objects/Bullet.java]: javadoc '@return impacted boolean true for when impact occurs'

File: core/src/main/java/com/github/integritygame/objects/Tank.java [hunk 7]
```diff
@@ -92,6 +114,23 @@ public void rotate(boolean clockwise){
     public Vector2 getCurrentPosition() {
         return new Vector2((position.x + (width / 2) + (side ? -18 : +18)), (position.y + (height / 2)) + 7);
     }
+    
+    /**
+     * Gets the end of the barrel of the tank's gun
+     */
```
Comment (jwilliamson15) [inline core/src/main/java/com/github/integritygame/objects/Tank.java]: javadoc @return line needed

File: core/src/main/java/com/github/integritygame/objects/Tank.java [hunk 8]
```diff
@@ -109,5 +148,52 @@ public void setTexture(String body, String turret){
...
+    public void setTankBody(Body tankBody) {
+        tankBody.setUserData(this);
+        this.tankBody = tankBody;
+    }
+    
+    
+    /**
+     * @return true if dead 
+     */
+    public boolean toggleByValue(boolean increase, int value){
```
Comment (jwilliamson15) [inline core/src/main/java/com/github/integritygame/objects/Tank.java]: method name needs clarifying such as isDead or checkForDeath. Java doc needs more detail

File: core/src/main/java/com/github/integritygame/util/GameManager.java [hunk 9]
```diff
@@ -66,11 +88,10 @@ public void render(){
             bullets.render(spriteBatch);
         spriteBatch.end();
 
-        //Also render the line from each tank to fire
-        //userA.getTank().renderShape(shapeRenderer);
```
Comment (jwilliamson15) [inline core/src/main/java/com/github/integritygame/util/GameManager.java]: remove if not used and/or needed

**Pull request merged**